teamcity: failed tests on master: testrace/TestNonRetryableErrorOnCommit

The following tests appear to have failed:

[#312876](https://teamcity.cockroachdb.com/viewLog.html?buildId=312876):

```
--- FAIL: testrace/TestNonRetryableErrorOnCommit (0.000s)
Race detected!

------- Stdout: -------
E170802 15:14:03.397796 187044 storage/replica_proposal.go:538  [n1,s1,r1/1:/M{in-ax}] could not load SystemConfig span: must retry later due to intent on SystemConfigSpan
E170802 15:14:03.574562 187055 storage/replica_proposal.go:538  [n1,s1,r4/1:/{System/tse-Max}] could not load SystemConfig span: must retry later due to intent on SystemConfigSpan
==================
Read at 0x00c42035ead8 by goroutine 122:
  github.com/cockroachdb/cockroach/pkg/sql_test.TestNonRetryableErrorOnCommit()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/txn_restart_test.go:1115 +0x312
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:657 +0x107

Previous write at 0x00c42035ead8 by goroutine 288:
  github.com/cockroachdb/cockroach/pkg/sql_test.TestNonRetryableErrorOnCommit.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/txn_restart_test.go:1104 +0x10e
  github.com/cockroachdb/cockroach/pkg/sql_test.(*CommandFilters).runFiltersInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:77 +0xcb
  github.com/cockroachdb/cockroach/pkg/sql_test.(*CommandFilters).(github.com/cockroachdb/cockroach/pkg/sql_test.runFiltersInternal)-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:108 +0x7c
  github.com/cockroachdb/cockroach/pkg/testutils/storageutils.(*ReplayProtectionFilterWrapper).run()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/storageutils/mocking.go:78 +0x129
  github.com/cockroachdb/cockroach/pkg/testutils/storageutils.(*ReplayProtectionFilterWrapper).(github.com/cockroachdb/cockroach/pkg/testutils/storageutils.run)-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/storageutils/mocking.go:46 +0x7c
  github.com/cockroachdb/cockroach/pkg/sql_test.(*CommandFilters).runFilters()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:70 +0xfe
  github.com/cockroachdb/cockroach/pkg/sql_test.(*CommandFilters).(github.com/cockroachdb/cockroach/pkg/sql_test.runFilters)-fm()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:189 +0x7c
  github.com/cockroachdb/cockroach/pkg/storage.evaluateCommand()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_command.go:182 +0xda3
  github.com/cockroachdb/cockroach/pkg/storage.evaluateBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:4791 +0x3ca
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).evaluateTxnWriteBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:4604 +0x2ce
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).evaluateProposalInner()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:4444 +0x156
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).evaluateProposal()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:2663 +0x147
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).requestToProposal()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:2620 +0x2f4
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).propose()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:2779 +0x21f
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).tryExecuteWriteBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:2524 +0x3e1
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).executeWriteBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:2378 +0xad
  github.com/cockroachdb/cockroach/pkg/storage.(*Replica).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/replica.go:1706 +0x7f9
  github.com/cockroachdb/cockroach/pkg/storage.(*Store).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:2578 +0x6c7
  github.com/cockroachdb/cockroach/pkg/storage.(*Stores).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/storage/stores.go:176 +0x265
  github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:792 +0x1fc
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:243 +0x104
  github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:803 +0x20b
  github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:820 +0xb7
  github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext.func1.1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:224 +0x728
  github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:237 +0x10b

Goroutine 122 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:697 +0x543
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:882 +0xaa
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:657 +0x107
  testing.runTests()
      /usr/local/go/src/testing/testing.go:888 +0x4e0
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:822 +0x1c3
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:222 +0x189
  main.main()
      github.com/cockroachdb/cockroach/pkg/sql/_test/_testmain.go:396 +0x20f

Goroutine 288 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/kv.(*grpcTransport).SendNext()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport.go:249 +0x34f
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1145 +0x6bc
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:394 +0x352
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendSingleRange()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:458 +0x1b7
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendPartialBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:940 +0x574
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).divideAndSendBatchToRanges()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:805 +0xcd5
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).Send()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:609 +0x3ed
  github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).tryAsyncAbort.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:714 +0x100
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTask.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:264 +0xf3
==================
I170802 15:06:08.665614 1 rand.go:74  Random seed: -835468156047126165

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/txn_restart_test.go
+++ b/pkg/sql/txn_restart_test.go
@@ -1096,24 +1096,17 @@ func TestNonRetryableErrorOnCommit(t *testing.T) {
 	s, sqlDB, _ := serverutils.StartServer(t, params)
 	defer s.Stopper().Stop(context.TODO())
 
-	hitError := false
-	cleanupFilter := cmdFilters.AppendFilter(
-		func(args storagebase.FilterArgs) *roachpb.Error {
-			if req, ok := args.Req.(*roachpb.EndTransactionRequest); ok {
-				if bytes.Contains(req.Key, []byte(keys.SystemConfigSpan.Key)) {
-					hitError = true
-					return roachpb.NewErrorWithTxn(fmt.Errorf("testError"), args.Hdr.Txn)
-				}
+	defer cmdFilters.AppendFilter(func(args storagebase.FilterArgs) *roachpb.Error {
+		if req, ok := args.Req.(*roachpb.EndTransactionRequest); ok {
+			if bytes.Contains(req.Key, []byte(keys.SystemConfigSpan.Key)) {
+				return roachpb.NewErrorWithTxn(errors.New(t.Name()), args.Hdr.Txn)
 			}
-			return nil
-		}, false)
-	defer cleanupFilter()
+		}
+		return nil
+	}, false)()
 
-	if _, err := sqlDB.Exec("CREATE DATABASE t"); !testutils.IsError(err, "pq: testError") {
+	if _, err := sqlDB.Exec("CREATE DATABASE t"); !testutils.IsError(err, t.Name()) {
 		t.Errorf("unexpected error %v", err)
-	}
-	if !hitError {
-		t.Errorf("expected to hit error, but it didn't happen")
 	}
 }
 

